lib/astarte_device_sdk/device_caching.c

```diff
@@ -78,6 +80,67 @@ static astarte_result_t append_property_to_string(introspection_t *introspection
  *         Global functions definitions         *
  ***********************************************/
 
+astarte_result_t astarte_device_caching_synchronization_get(bool *sync)
+{
+    astarte_result_t ares = ASTARTE_RESULT_OK;
+    astarte_kv_storage_t kv_storage = { 0 };
+
+    ASTARTE_LOG_DBG("Getting stored synchronization");
+
+    ares = open_kv_storage(SYNCHRONIZATION_NAMESPACE, &kv_storage);
+    if (ares != ASTARTE_RESULT_OK) {
+        ASTARTE_LOG_ERR("Init error for synchronization cache: %s.", astarte_result_to_name(ares));
+        goto exit;
+    }
+
+    bool read_sync = false;
+    size_t read_sync_size = sizeof(read_sync);
+
+    ares = astarte_kv_storage_find(&kv_storage, SYNCHRONIZATION_KEY, &read_sync, &read_sync_size);
+    if (ares == ASTARTE_RESULT_NOT_FOUND) {
+        ASTARTE_LOG_INF("No previous synchronization with Astarte present.");
+        goto exit;
+    }
+    if (ares != ASTARTE_RESULT_OK) {
+        ASTARTE_LOG_ERR("Fetch error for cached introspection: %s.", astarte_result_to_name(ares));
+        goto exit;
+    }
+
+    if (!read_sync) {
+        ASTARTE_LOG_INF("No previous synchronization with Astarte present.");
+        *sync = false;
+    } else {
+        *sync = true;
+    }
```

Comment:
What do you think to leave only the log in the if block?

```suggestion
    if (!read_sync) {
        ASTARTE_LOG_INF("No previous synchronization with Astarte present.");
    }
    *sync = read_sinc;
```